queries.py

```diff
@@ -1,160 +1,161 @@
 import os
-import json
 import pandas as pd
 from databricks import sql
 
 SERVER_HOSTNAME = os.getenv("SERVER_HOSTNAME")
 HTTP_PATH = os.getenv("HTTP_PATH")
 ACCESS_TOKEN = os.getenv("ACCESS_TOKEN")
+PUBLIC_ONLY = os.getenv("PUBLIC_ONLY", "False").lower() in ("true", "1", "yes")
 
+class QueryService:
+
+    _instance = None
+
+    @staticmethod
+    def get_instance():
+        if QueryService._instance is None:
+            QueryService._instance = QueryService()
+        return QueryService._instance
+
+    def __init__(self):
+        self.country_whitelist = None
+
+    def execute_query(self, query):
+        """
+        Executes a query and returns the result as a pandas DataFrame.
+        """
+        conn = sql.connect(
+            server_hostname=SERVER_HOSTNAME,
+            http_path=HTTP_PATH,
+            access_token=ACCESS_TOKEN,
+        )
+        cursor = conn.cursor()
+        cursor.execute(query)
+        df = cursor.fetchall_arrow().to_pandas()
+        cursor.close()
+        conn.close()
+        return df
+
+    def _initialize_country_whitelist(self):
+        if PUBLIC_ONLY and self.country_whitelist is None:
+            query = """
+                SELECT country_name
+                FROM boost.data_availability
+                WHERE boost_public = 'Yes'
+            """
+            self.country_whitelist = self.execute_query(query)["country_name"].tolist()
+
+    def _apply_country_whitelist_filter(self, df):
+        if self.country_whitelist is not None and 'country_name' in df.columns:
```

Comment:
might be safer to check PUBLIC_ONLY=YES before returning the unfiltered df.

Reply:
What I was concerned about was that when the condition
`self.country_whitelist is not None and 'country_name' in df.columns:` is false for some reason (even when PUBLIC_ONLY is true), we will be exposing non-public country information to external users.